bruteforce/2309.일곱 난쟁이/2.kt

```diff
@@ -0,0 +1,50 @@
+import java.io.BufferedReader
+import java.io.BufferedWriter
+import java.io.InputStreamReader
+import java.io.OutputStreamWriter
+
+// 백트래킹 기법 풀이
+
+var flag = false
+fun main() {
+    val br= BufferedReader(InputStreamReader(System.`in`))
+    val bw = BufferedWriter(OutputStreamWriter(System.out))
+    val list = mutableListOf<Int>().apply {
+    repeat(9)    {
+        add(br.readLine().toInt())
+        }
+    }
+    list.sort()
+    val visited = mutableListOf<Boolean>().apply {
+        repeat(9){
+            add(false)
+        }
+    }
+    backTracking(0, 0, list, visited)
+   
+    for (i in 0 .. 8){
+        if(visited[i]) {
+            bw.write("" + list[i] + "\n")
+        }
+    }
+    bw.flush()
+    bw.close()
+    br.close()
+}
+
+fun backTracking(sum: Int = 0,
+                 num: Int,list: MutableList<Int>,
+                 visited: MutableList<Boolean>) {
+    if (num == 7){
+        if(sum == 100) {flag = true }
+        return
+    }
+    for (i in 0 .. 8){
+        if(!visited[i]){
+            visited[i] = true
+            backTracking(sum+list[i], num+1, list, visited)
```

Comment:
```suggestion
            backTracking(sum + list[i], num + 1, list, visited)
```
이런 거 보고도 그냥 넘어가고 싶구만

Reply:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ 선생님... 이건 알고리즘이라구요!! ㅜㅜ 
다음부터는 cmd +option + L 하겠습니다.. ㅜ